### PR TITLE
Add support for PerkinElmer Vectra/QPTIFF files

### PR DESCRIPTION
**Pull request by melissalinkert**
See https://trello.com/c/bRR5zfc7/200-add-support-for-perkinelmer-vectra-qptiff-files

This will require a minor release, as it introduces a new reader and adds an API method to ```loci.formats.gui.AWTImageTools```.

Vectra/QPTIFF is a pyramid TIFF format used for both whole slide scans (fluorescence or brightfield) and multispectral images with smaller (< 2k x 2k) XY dimensions.  Whole slide scans typically use the file extension ```.qptiff``` while other data types use the extension ```.tif```.

To test use, the sample files from https://github.com/openmicroscopy/data_repo_config/pull/248.  The image and series counts reported by ```showinf``` should be such that the sum of the image counts matches the number of IFDs (i.e. ```tiffinfo /path/to/file | grep -c "TIFF Directory"```).  Verify that opening each file in ImageJ results in at least 2 series, each with a valid thumbnail in the ```Bio-Formats Series Options``` window.  We do not have other software that opens these files, so testing is mainly verifying that there is nothing that looks obviously wrong or strange.

b9c3284 fixes long-standing issues with thumbnails for floating point images in ImageJ.  It is needed to improve the user experience when working with these files in ImageJ, as many of the multispectral datasets contain 32-bit float data, and all have at least 2 series.  I can split this commit into a separate PR if desired.

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-DEV-merge-push#1221](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/1221/). See the [console output](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/1221/consoleText) for more details.
Possible conflicts:
  - PR #2959 rfinck 'Add reader for IonpathMIBI TIFF'
    - components/formats-api/src/loci/formats/readers.txt

--conflicts

**Comment by melissalinkert** [comment]
cd08051 should hopefully fix the merge conflict so that we can see if/how this affects builds.

**Comment by dgault** [comment]
The code changes here look to be good. For testing I have gone through each of the files in https://github.com/openmicroscopy/data_repo_config/pull/248, firstly comparing the output of showinf with that of tiffinfo. I then opened each image in ImageJ and verified that each series had the correct image count and that the images displayed correctly, both in thumbnail form and for each available channel when opened.

The fix for https://github.com/openmicroscopy/bioformats/commit/b9c3284b8664bd6d23cd3c0002456d4177eb5d7b also looks good and with this commit included there are no issues with ImageJ thumbnails that had previously been seen.

Overall this and the associated config PR look good to merge.

File: components/formats-gpl/src/loci/formats/in/VectraReader.java [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+  /* (non-Javadoc)
+   * @see loci.formats.FormatReader#isThisType(java.lang.String, boolean)
+   */
+  @Override
+  public boolean isThisType(String name, boolean open) {
+    if (!open) {
+      return checkSuffix(name, "qptiff");
+    }
+    RandomAccessInputStream stream = null;
+    try {
```
Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/VectraReader.java]: Channeling @mtbc here, this could probably be converted into a `try-with-resources`

Comment (melissalinkert) [reply components/formats-gpl/src/loci/formats/in/VectraReader.java]: Addressed in https://github.com/openmicroscopy/bioformats/pull/3060

**Pull request merged**